CONTRIBUTING.md

```diff
@@ -0,0 +1,73 @@
+# Contributing to this repository
```

Comment:
Are you suggesting that we add a copy of this to each and every repo? If so, we should keep a master copy somewhere in case things need to be changed. If so, the first question is where do we keep this - gitlab or github? I'd go for gitlab, personally, as we tend to use that for most things, and only really use github for stackhpc/public things

Reply:
Initially yes, but I then had exactly the same thought as you, just this morning.

Totally agree on the master copy. I guess there's a few things we could do...

- master copy in a 'guidelines' repo, and submodule that in each other repo. Feels massively overkill to me. Also lots of folks hate submodules
- master copy in a 'guidelines' repo, and some kind of CI hook to propose changes to all other repos. Also feels overkill
- any other form of master repo/scripted/automated hook thingy...
- Keep it in confluence, and just reference/hyperlink to it in each repo.

last one would work for me, interested to see other folks opinions

Reply:
I think a standard template repo wouldn't be overkill... well it would if it were _just_ for this file, but there'll no doubt be others, such as README.md, LICENSE, maybe .gitignore.

If Python, we have a fairly standard version way of doing versions in one of the __init__.py files, maybe... then there's .flake8, .tox, etc

A single repo to store this default template repo wouln't be so bad, but it could also just go in confluence. Either way... ¯\_(ツ)_/¯

Reply:
I'd argue that README and possibly LICENSE would likely be different across repos. The other stuff I'd expect to be more like repo scaffolding you'd use on repo setup - still very useful, but a different use case to needing to manage a potentially changing file across 70+ repos.

Again though, happy to go with consensus

Reply:
I propose putting it in confluence to start with as that's the lowest barrier, and we can move it later if that doesn't work for us.

Reply:
+1 for confluence to start